sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/common/IOITMetrics.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.common;
+
+import com.google.cloud.Timestamp;
+import java.util.Collection;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.testutils.NamedTestResult;
+import org.apache.beam.sdk.testutils.metrics.MetricsReader;
+import org.apache.beam.sdk.testutils.publishing.BigQueryResultsPublisher;
+import org.apache.beam.sdk.testutils.publishing.ConsoleResultPublisher;
+
+public class IOITMetrics {
+
+  private final Set<Function<MetricsReader, NamedTestResult>> metricSuppliers;
+  private final PipelineResult result;
+  private final String namespace;
+
+  public IOITMetrics(
+      Set<Function<MetricsReader, NamedTestResult>> metricSuppliers,
+      PipelineResult result,
+      String namespace) {
+    this.metricSuppliers = metricSuppliers;
+    this.result = result;
+    this.namespace = namespace;
+  }
+
+  public void publish(String bigQueryDataset, String bigQueryTable) {
+    String uuid = UUID.randomUUID().toString();
```

Comment:
This should be passed to IOITMetrics from the test class (uuid, timestamp), right?